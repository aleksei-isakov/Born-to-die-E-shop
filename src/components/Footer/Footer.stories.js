import Footer from './Footer.vue';

export default {
  title: 'Footer',
  component: Footer
};

const FooterTemplate = () => ({
  components: { Footer },
  template: `<Footer/>`
});

export const Default = FooterTemplate.bind({});
