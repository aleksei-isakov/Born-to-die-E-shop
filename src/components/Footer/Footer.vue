<template>
  <div class="footer-wrapper">
    <span>EPAM Systems &copy; 2022</span>
  </div>
</template>

<script>
export default {
  name: 'Footer'
};
</script>

<style lang="scss" scoped>
@import '@/scss/CustomVariables.scss';
.footer-wrapper {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  background-color: $background-light-grey;
  height: 40px;
  margin-top: auto;
}
</style>
