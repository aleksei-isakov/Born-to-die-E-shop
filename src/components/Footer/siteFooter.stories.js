import siteFooter from './siteFooter.vue';

export default {
  title: 'Footer',
  component: siteFooter
};

const FooterTemplate = () => ({
  components: { siteFooter },
  template: `<site-footer/>`
});

export const Default = FooterTemplate.bind({});
