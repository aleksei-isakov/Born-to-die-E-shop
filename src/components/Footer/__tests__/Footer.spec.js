import { shallowMount, createLocalVue } from '@vue/test-utils';
import Footer from '../../Footer/Footer.vue';

let wrapper;

beforeEach(() => {
  wrapper = shallowMount(Footer, {});
  expect(wrapper.is(Footer)).toBeTruthy();
});

afterEach(() => {
  wrapper.destroy();
});

describe('Footer', () => {
  test("has 'footer-wrapper' class", () => {
    expect(wrapper.contains('.footer-wrapper')).toBe(true);
  });
});

describe('Footer', () => {
  test('is a Vue instance', () => {
    expect(wrapper.isVueInstance).toBeTruthy();
  });
});

describe('Footer', () => {
  it('renders a valid snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
