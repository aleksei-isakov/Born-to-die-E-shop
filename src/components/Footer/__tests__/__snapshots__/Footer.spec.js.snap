// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders a valid snapshot 1`] = `
<div
  class="footer-wrapper"
>
  <span>
    EPAM Systems Â© 2022
  </span>
</div>
`;
